### ISSUE TITLE
Feature Request: Alternative to docker on Windows: Ready-to-use pre-installed VM

### ISSUE THREAD

**Issue by AnonyPla** [freedomofpress/dangerzone#125]
Hello,

As you may or may not know, Docker-Desktop on Windows (the latest version) requires some specific features turned on to function: 

On Windows Home:

- WSL2
- Virtual Machine Platform

On Windows Pro:

- Hyper-V
- Containers

The problem is that these features interfere with (or just break) Virtualization software such as Virtualbox or VMWare degrading their performance significantly. Therefore, to use Dangerzone on Windows, you have to accept these features which degrade the performance of other Virtual Machines severely. It also makes some VMs just unusable.

I would like to suggest a new approach in the form of providing a ready-to-use Virtualbox Linux VM with Dangerzone pre-installed and Virtualbox Guest Additions enabled by default for draf & drop functionalities.

- This will allow Windows (or any other platform users) to just run/use Dangerzone without the docker requirement.
- This will reduce the attack surface of any malicious software even further by running Dangerzone within a VM in addition to docker containers.
- Users can simply drag & drop files from their Host OS (Windows for instance) to the Dangerzone VM and then use Dangerzone on the VM for processing.

This wouldn't require a lot of work and you could provide a minimalist XFCE Linux VM (Ubuntu or Debian for instance) .ova file for people to import in Virtualbox. 

This approach:

- Adds security by adding a Virtualization layer on top (in case a malicious file would compromise Dangerzone)
- Removes the need for any OS features enabling on Windows (Hyper-V...)
- Removes the need to install Docker on your main OS
- Provides a ready-to-use tested controlled environment where you know it will work and so reduce the amount of install/setup related issues and support required
- Removes the need to provide multi-platform releases in favor of a single easy-to-run VM on any OS where Virtualbox runs

What do you think?

Regards

AnonyPla

**Comment by yveszoundi** [comment]
@AnonyPla, that's exactly how I use my "[variation/fork of Dangerzone, dangerzone-rust](https://github.com/rimerosolutions/dangerzone-rust)": work and home usage. Maybe #110 is a step in that direction for Dangerzone.

I convert documents from a Web interface running within a "dangerzone-rust Virtual Machine" (Debian on QEMU)
, as I stopped tolerating VirtualBox atrocious performance on any operating system). Including some of the reasons that you mentioned, there's a use-case for "a shared online service" (home network or intranet).

![Web interface](https://raw.githubusercontent.com/rimerosolutions/dangerzone-rust/main/dangerzone-httpserver/images/screenshot-web.png) 

My "dangerzone-rust virtual machine" has the following characteristics:
- The memory is capped to 1 GB of RAM with only 1 CPU allocated
- Internet connectivity is disabled, only SSH and "Web conversion requests" are allowed (host & other VMs)
  - No graphical interface, as there's the Web interface to convert documents: I don't need the desktop experience
  - Occasionally I need to convert documents in batch, but I do that from a shell script and the "dangerzone-cli"

**Comment by yveszoundi** [comment]
I experimented briefly with the concept of a Live CD, I'll likely adopt that approach instead of applying [ansible](https://docs.ansible.com/ansible/latest/user_guide/playbooks_intro.html) playbooks. 

TL;DR  Find below a summary for my experimentation, in case it can help "someone" implement this feature request.

### Overview

- [The live CD is based on Debian bullseye](https://willhaley.com/blog/custom-debian-live-environment/), packaging podman and a `dangerzone-httpserver` component.
- There's no GUI, to keep the system minimal and also because there's a "Web interface" to `dangerzone`
- There's no internet connectivity, only SSH and access to the `dangerzone-httpserver` is allowed (iptables)

### User journey
- The user downloads the live CD from the relevant [Github release page](https://github.com/rimerosolutions/dangerzone-rust/releases/tag/0.0.2): `dangerzone-livecd.iso`
- The user boots the live CD ISO image with the virtualization tool of their choice (qemu, virtualbox, etc.)
- The user starts converting "potentially suspicious" documents from their browser
  - Access the `dangerzone-rust` Web UI served from the live CD `http://ip_address:13000/`
  - In the Web UI, upload the document to convert
  - Once the conversion is completed, download the resulting PDF from the Web UI: 
    - The original document and its resulting PDF are deleted immediately from the server upon download.
    - Possibly, batch conversions will be supported in the Web interface at some point (drag & drop -> bulk convert)

**Comment by deeplow** [comment]
Let's continue this conversation on #118, since whichever solution we come up with should work for both Mac and Windows.
